src/main/java/discord4j/discordjson/json/RoleData.java

```diff
@@ -22,11 +24,11 @@
 
     boolean hoist();
 
-    int position();
+    Possible<Optional<Integer>> position();
```

Comment:
I need to review these field type changes (position and managed). https://discord.com/developers/docs/topics/permissions#role-object don't indicate this is a nullable value, optional field.

Maybe a different RoleData class is a better fit. if RoleData is our full Role object, but this new RoleTemplateData partially covers it. My suggestion is to:

- Create a new RoleDataFields interface with fields common to both (check existing interfaces ending in -Fields)
- Make RoleData extend RoleDataFields, and adding the position + managed fields
- Create a new RoleTemplateData that extends RoleDataFields, only adding the static builder method standard to our data objects

Let me know what you think about it and if it's right or not. Also @darichey @danthonywalker please take a look

Reply:
I'll create a RoleTemplateData that extends RoleData. When using a template the position object didn't exist so i thought Discord changed how RoleData works.

Reply:
RoleData will have the same fields and you want to exclude `position` and `managed` from RoleTemplateData, so make sure the latter extends an interface with only the common fields.

Reply:
Yes i just realised that